src/python/pants/option/arg_splitter.py

```diff
@@ -213,12 +214,23 @@ def assign_flag_to_scope(flg: str, default_scope: str) -> None:
         if not goals and not self._help_request:
             self._help_request = NoGoalHelp()
 
+        # When we remove this deprecation, we will remove passthru "owners" entirely: all scopes
+        # which have passthrough args will receive them.
+        deprecated_conditional(
+            lambda: bool(passthru) and len(passthru_owners) > 1,
+            removal_version="1.31.0.dev0",
+            entity_description="Ambiguous scopes for passthrough args",
+            hint_message="In future, all scopes which accept passthrough args (arguments after --) "
+            f"will receive them, so passing multiple scopes (in this case: {passthru_owners}) "
+            "while using passthrough args is no longer recommended.",
```

Comment:
Yea, I think that that makes sense. An easy way to make things less confusing.

Reply:
Fantastic. I'll try to think of other changes like this to address my concerns.

As discussed over DM, I ran into an issue in the `release.sh` script not being able to use the `--` syntax because I couldn't get an array of strings to work properly with `--run-args`. So, I'm coming around to the value of this syntax, particularly if we can find small changes like this.

Reply:
Did this in the latest revision.